wherehows-backend/app/controllers/Application.java

```diff
@@ -13,12 +13,69 @@
  */
 package controllers;
 
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import play.Logger;
+import play.Play;
 import play.mvc.Controller;
 import play.mvc.Result;
+import wherehows.dao.DaoFactory;
 
 
 public class Application extends Controller {
 
+  private static final String DB_WHEREHOWS_URL = Play.application().configuration().getString("db.wherehows.url");
+  private static final String DB_WHEREHOWS_USERNAME =
+      Play.application().configuration().getString("db.wherehows.username");
+  private static final String DB_WHEREHOWS_PASSWORD =
+      Play.application().configuration().getString("db.wherehows.password");
+
+  public static final DaoFactory daoFactory = createDaoFactory();
+
+  private static DaoFactory createDaoFactory() {
+    try {
+      String className =
+          Play.application().configuration().getString("dao.factory.class", DaoFactory.class.getCanonicalName());
+      Class factoryClass = Class.forName(className);
+      Constructor<? extends DaoFactory> ctor = factoryClass.getConstructor();
+      DaoFactory daoFactory = ctor.newInstance();
+      daoFactory.setEntityManager(createEntityManager());
+      return daoFactory;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static EntityManager createEntityManager() {
+    Logger.info("start the application, load info: " + DB_WHEREHOWS_URL + "\n" + DB_WHEREHOWS_USERNAME + "\n"
+        + DB_WHEREHOWS_PASSWORD);
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put("hibernate.connection.provider_class", "org.hibernate.hikaricp.internal.HikariCPConnectionProvider");
+    properties.put("hibernate.hikari.dataSourceClassName", "com.mysql.jdbc.jdbc2.optional.MysqlDataSource");
+    properties.put("hibernate.hikari.dataSource.url", DB_WHEREHOWS_URL);
+    properties.put("hibernate.hikari.dataSource.user", DB_WHEREHOWS_USERNAME);
+    properties.put("hibernate.hikari.dataSource.password", DB_WHEREHOWS_PASSWORD);
+    properties.put("hibernate.hikari.dataSource.cachePrepStmts", "true");
+    properties.put("hibernate.hikari.dataSource.prepStmtCacheSize", "250");
+    properties.put("hibernate.hikari.dataSource.prepStmtCacheSqlLimit", "2048");
+    properties.put("hibernate.hikari.minimumIdle", "5");
+    properties.put("hibernate.hikari.maximumPoolSize", "10");
+    properties.put("hibernate.hikari.idleTimeout", "30000");
+    properties.put("hibernate.show_sql", "false");
+    properties.put("hibernate.dialect", "MySQL5");
+    properties.put("hibernate.jdbc.batch_size", "100");
+    properties.put("hibernate.order_inserts", "true");
+    properties.put("hibernate.order_updates", "true");
+
+    EntityManagerFactory factory = Persistence.createEntityManagerFactory("default", properties);
```

Comment:
I think it might be better to make the Factory public static final. And each DAO transaction will create a new EntityManager and close it after use.